Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardLibrary.c

```diff
@@ -78,18 +78,23 @@ GetBootGuardInfo (
     BootGuardBootStatus  = *(UINT64 *) (UINTN) (TXT_PUBLIC_BASE + R_CPU_BOOT_GUARD_BOOTSTATUS);
     DEBUG ((DEBUG_INFO, "Boot Guard Boot Status = %llx\n", BootGuardBootStatus));
 
+    DEBUG ((DEBUG_INFO, "HeciBaseAddress = %llx\n", HeciBaseAddress));
     ///
     /// Read ME FWS Registers
     ///
-    MeFwSts4 = MmioRead32 (HeciBaseAddress + R_ME_HFS_4);
-    DEBUG ((DEBUG_INFO, "ME FW STS 4 = %x\n", MeFwSts4));
+    if (HeciBaseAddress != 0) {
+      MeFwSts4 = MmioRead32 (HeciBaseAddress + R_ME_HFS_4);
+      DEBUG ((DEBUG_INFO, "ME FW STS 4 = %x\n", MeFwSts4));
+    } else
+      MeFwSts4 = 0x0;
 
     ///
     /// Check Bit 12 in ME FWSTS4 to check if TPM_DISCONNECT_ALL bit is set
     /// or ENF Shutdown path is taken by ME FW.
     /// Also Identify any failures in ACM
     ///
-    if (((MeFwSts4 & (B_TPM_DISCONNECT | B_BOOT_GUARD_ENF_MASK)) != 0) ||
+    if ((((MeFwSts4 & (B_TPM_DISCONNECT | B_BOOT_GUARD_ENF_MASK))
+        && HeciBaseAddress) != 0) ||
```

Comment:
HeciBaseAddress need not be checked as MeFwSts4 will be zero.

Reply:
done